ls1intum/Artemis
# `General`: Show error messages for empty courses during export

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
Closes #7288. Empty Courses seemed to be loading forever during archiving and showed a loading spinner, because the exercise task had been skipped and therefore the "complete" message was not sent.

### Description
This PR adds messages in the case of error, so that the client knows whenever the archiving fails and shows an error alert.

### Steps for Testing

Prerequisites:
- 1 Instructor


1. Log in to Artemis
2. Navigate to Course Administration
3. Create New Course with End Date before today.
4. Click the "Archive Course" Button and verify that there is an error displayed.

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with a Programming Exercise

1. Log in to Artemis
2. Participate in the exam as a student
3. Make sure that the UI of the programming exercise in the exam mode stays unchanged. You can use the [exam mode documentation](https://ls1intum.github.io/Artemis/user/exam_mode/) as reference.
5. ...

### Review Progress






#### Code Review
#### Manual Tests


### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [CourseExamExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5603/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/CourseExamExportService.html) | 91% | ✅ |
| [course-exam-archive-button.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5603/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/components/course-exam-archive-button/course-exam-archive-button.component.ts.html) | 87.93% | ✅|
### Screenshots
![image](https://github.com/ls1intum/Artemis/assets/78978542/af008ae0-9c9d-4759-b9d0-33744cceb058)
